components/Funnels/AnalysisButton.tsx

```diff
@@ -12,13 +11,13 @@ const AnalysisButton = ({
   containerClasses?: string;
 }) => {
   const { username, funnelType } = useFunnelAnalysisProvider();
-  const { handleAnalyze: handleTwitterAnalysis } = useTwitterAnalysisProvider();
   const { handleAnalyze: handleSpotifyAnalysis } = useSpotifyAnalysisProvider();
   const { openAgentSocket } = useAgentSocketProvider();
 
   const handleClick = () => {
     if (funnelType === Funnel_Type.TIKTOK) openAgentSocket(Funnel_Type.TIKTOK);
-    if (funnelType === Funnel_Type.TWITTER) handleTwitterAnalysis();
+    if (funnelType === Funnel_Type.TWITTER)
+      openAgentSocket(Funnel_Type.TWITTER);
```

Comment:
Could you reduce all Agent calls to one call?

Is this the same call with less code?
```
if (funnelType === Funnel_Type.TIKTOK || funnelType === Funnel_Type.TWITTER) {
    openAgentSocket(funnelType);
}
```